lib/dai-plugin-mcd/src/ManagedCdp.js

```diff
@@ -32,6 +40,74 @@ export default class ManagedCdp {
     return art.times(rate).shiftedBy(-27);
   }
 
+  async getCollateralizationRatio() {
+    const [collateral, debt] = await Promise.all([
+      this.getCollateralValue(),
+      this.getDebtValue()
+    ]);
+    assert(debt.gt(0), 'Debt cannot be zero');
+    return collateral.div(debt).toNumber();
+  }
+
+  async getLiquidationPrice() {
+    const [liquidationRatio, debt, collateral] = await Promise.all([
+      this._cdpType.getLiquidationRatio(),
+      this.getDebtValue(),
+      this.getCollateralAmount()
+    ]);
+    assert(collateral.gt(0), 'Collateral cannot be zero');
+    const ratio = createCurrencyRatio(USD, this._cdpType.currency);
+    return ratio(
+      debt
+        .times(liquidationRatio)
+        .div(collateral)
+        .toNumber()
+    );
+  }
+
+  async isSafe() {
+    const [liqPrice, collPrice] = await Promise.all([
+      this.getLiquidationPrice(),
+      this._cdpType.getPrice()
+    ]);
+    return collPrice.gte(liqPrice);
+  }
+
+  async getMinCollateralAmount() {
+    const [debt, liquidationRatio, price, collateral] = await Promise.all([
+      this.getDebtValue(),
+      this._cdpType.getLiquidationRatio(),
+      this._cdpType.getPrice(),
+      this.getCollateralAmount()
+    ]);
+    const lockedCollateral = this._cdpType.currency(
+      debt.times(liquidationRatio).div(price.toNumber())
+    );
+    return lockedCollateral.toNumber() > collateral.toNumber()
+      ? collateral
+      : lockedCollateral;
+  }
+
+  async getCollateralAvailable() {
+    const [collateral, lockedCollateral] = await Promise.all([
+      this.getCollateralAmount(),
+      this.getMinCollateralAmount()
+    ]);
+    return collateral.minus(lockedCollateral);
+  }
+
+  async daiAvailable() {
+    const [collateral, liquidationRatio, debt] = await Promise.all([
+      this.getCollateralValue(),
+      this._cdpType.getLiquidationRatio(),
+      this.getDebtValue()
+    ]);
+    const dai = USD(collateral.toNumber())
```

Comment:
given the unit change i suggested to `getCollateralValue`, the units should work out here without any wrapping. USD / (USD/DAI) - DAI = DAI. `CdpType.getLiquidationRatio` is missing the USD_DAI type though, so that will have to be added.